module/zfs/zil.c

```diff
@@ -3105,6 +3103,38 @@ zil_commit_impl(zilog_t *zilog, uint64_t foid)
 	zil_free_commit_waiter(zcw);
 }
 
+/*
+ * Called as part of zil_sync once the replay succeeded successfully.
+ *
+ * For backwards-compatibility reasons, brand new TX_* records need to have a
+ * feature flag associated with them so that a system will not be confused when
+ * importing a pool with unknown TX_* types. These features are only activated
+ * while the ZIL contains a log entry containing the new TX_* type, and needs
+ * to be disabled when the ZIL has been applied and cleared (so that the pool
+ * can be imported onto other systems after a clean 'zfs export').
```

Comment:
Have you tested this? IIRC when we looked at ZILSAXATTR, the feature would still be reported as `enabled` (even though there were no datasets using it). Maybe I missed the place where you can transition a feature to `disabled` if it has a refcount of 0?

Reply:
Sorry, I meant "disactivated". I wrote this comment before I implemented the rest so the word usage wasn't correct. As far as I know, it is not possible (nor is it meant to be possible) for you to deactivate a feature in ZFS. I have not yet tested all of the new bits on my machine (I was having issues with my LXD VM setup -- the last time I did any significant testing was before all of the feature flag changes, so I will sit down and do some proper local testing before I ask for a proper review).

Reply:
I've done some tests (there was a mistake in the first version I had of this) and now it seems that the feature only gets deactivated _sometimes_ after a clean export. Looking into it...

Reply:
Interesting -- it seems like `zl_destroy_txg` is not being set when you do a clean export (I don't know how to trigger it as I did before I started debugging it). This is even if I do `zpool freeze` beforehand to do an unclean export + import loop.